app/controllers/escorts_controller.rb

```diff
@@ -16,14 +16,28 @@ def create
   def show
   end
 
+  def confirm_cancel
+    @form = Forms::Escort.new(escort)
+  end
+
+  def cancel
+    @form = Forms::Escort.new(escort)
+    if @form.validate(escort_params)
+      escort.cancel!(current_user, escort_params[:cancelling_reason]) unless escort.issued? || escort.cancelled?
```

Comment:
This logic `escort.issued? || escort.cancelled?` should probably be hidden in a method `can_be_cancelled?` so that if the logic changes we don't need to replicate it wherever its being used.